kv: de-flake TestTxnCoordSenderEndTxn

tbg opened the pull request:
It was not cognizant of the fact that a push target timestamp is taken
off the local clock and not computed from the pushee.

Fixes #6574.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/6577)

<!-- Reviewable:end -->

tamird commented:
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

*[kv/txn_coord_sender_test.go, line 416 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6577#-KHGFZ9bdPYQ5eMoHxce:-KHGFZ9cexvrKiBl-KXM:224227467) ([raw file](https://github.com/cockroachdb/cockroach/blob/a23ecde1bc6f7a632c768e9efe5a4c2dcfd0b9aa/kv/txn_coord_sender_test.go#L416)):*
> ```Go
> 			t.Fatal(pErr)
> 		} else {
> 			pushedTimestamp.Forward(txn.Timestamp)
> ```

since the other arm of the `if` is a `Fatal`, isn't this the same as `pushedTimestamp = txn.Timestamp` (without the `conflictTxn.Proto.Timestamp` initialization)?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6577)*
<!-- Sent from Reviewable.io -->

vivekmenezes commented:
LGTM. Thanks for fixing it!

tbg committed:
kv: de-flake TestTxnCoordSenderEndTxn

It was not cognizant of the fact that a push target timestamp is taken
off the local clock and not computed from the pushee.

tschottdorf commented:
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[kv/txn_coord_sender_test.go, line 416 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6577#-KHGFZ9bdPYQ5eMoHxce:-KHI3so0xBS-b7PHno1B:446643589) ([raw file](https://github.com/cockroachdb/cockroach/blob/a23ecde1bc6f7a632c768e9efe5a4c2dcfd0b9aa/kv/txn_coord_sender_test.go#L416)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> since the other arm of the `if` is a `Fatal`, isn't this the same as `pushedTimestamp = txn.Timestamp` (without the `conflictTxn.Proto.Timestamp` initialization)?</details>
Yep. Simplified.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6577)*
<!-- Sent from Reviewable.io -->

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

<details><summary><i>Previously, vivekmenezes wrote…</i></summary>
> LGTM. Thanks for fixing it!</details>


---

Reviewed 1 of 1 files at r2.
Review status: :shipit: all files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6577#-:-KHI7mmzbTLVifR4sXC_:1417088721)*
<!-- Sent from Reviewable.io -->
